File: typescript/packages/bench/src/rules/discountCeilingRule.ts
/** Cardinality fixture: nominal (class, `implements`) implementor of RuleContract. */

import type { RuleContract } from "../contracts/ruleContract";
import type { Dataset } from "../dataset";

export class DiscountCeilingRule implements RuleContract {
    readonly key = "discount-ceiling";

    evaluate(data: Dataset): boolean {
        return data.revenueCents() > 0;
    }
}

File: rust/crates/atelier-bench/src/rules/discount_ceiling_rule.rs
//! Rule 12/24 (textual half).

use crate::contracts::rule_contract::RuleContract;
use crate::dataset::Dataset;

pub struct DiscountCeilingRule;

impl RuleContract for DiscountCeilingRule {
    fn evaluate(&self, data: &Dataset) -> bool {
        !data.invoices.is_empty()
    }
}
